### PR TITLE
Improved mailer tests

### PR DESCRIPTION
**Pull request by prakharjoshi**
Improved functional and unit tests for easyforms.

### CODE REVIEW

**Comment by davilima6** [comment]
Lgtm. The Travis question above is not valid anymore.

@pbauer, @thet, @tomgross, @smcmahon: could any of you give a last look and merge?

File: collective/easyform/actions.py [hunk 1]
```diff
@@ -261,7 +261,8 @@ def get_addresses(self, fields, request, context, from_addr=None, to_addr=None):
         if hasattr(self, 'to_field') and self.to_field:
             recip_email = fields.get(self.to_field, None)
         if not recip_email:
-            recip_email = self.recipient_email
+            if self.recipient_email != '':
```
Comment (tomgross) [inline collective/easyform/actions.py]: What happens if self.recipient_email **is** empty?

File: collective/easyform/tests/testCustomScript.py [hunk 2]
```diff
@@ -196,34 +194,33 @@ def createScript(self):
...
+    #     form = view.form_instance
+    #     # form.update()
+    #     errors = form.widgets.errors
+    #     self.assertEqual(len(errors), 1)
+    #     self.assertEqual(errors[0].message, 'Please enter more text')
+    #     data, errors = form.extractData()
+    #     self.assertEqual(len(errors), 0)
+
+    #     errors = form.processActions(data)
+    #     self.assertEqual(errors, {'comments': 'Please enter more text'})
```
Comment (tomgross) [inline collective/easyform/tests/testCustomScript.py]: I'm not a fan of commenting out code. We have a VCS for that. ;-)

File: collective/easyform/tests/testMailer.py [hunk 3]
```diff
@@ -480,9 +480,13 @@ def testNoRecipient(self):
             topic='test subject', replyto='test@test.org', comments='test comments')
 
         self.messageText = ''
-        mailer.onSuccess(request.form, request)
-        self.assertTrue(
-            'mdummy@address.com' in self.mto
+        try:
```
Comment (tomgross) [inline collective/easyform/tests/testMailer.py]: assertRaises can be used as context manager (at least with Python 2.7 and/or unittest2) which is probably a bit nicer to read: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises

**Comment by tomgross** [comment]
LGTM Thanks @prakharjoshi

**Pull request merged**